source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+The following sections use a ``cursor`` variable, which is a cursor that
+retrieved all the documents in a collection with the ``Find()`` function.
```

Comment:
I agree with the first two and half of the third point-- For the third, I think it's ok to omit "returned from a find operation" because the snippet shows the `Find()` call.